contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -17,297 +17,223 @@
-///    member delegates stake, we *increase* their Shadow Balance and the Total Shadow Balance of the pool.
-///
-/// 2. When a member withdraws a portion of their reward, their realized balance increases but their ownership
-///    within the pool remains unchanged. Thus, we simultaneously *decrease* their Real Balance and
-///    *increase* their Shadow Balance by the amount withdrawn. The cumulative balance decrease and increase, respectively.
-///
-/// 3. When a member undelegates, the portion of their reward that corresponds to that stake is also withdrawn. Thus,
-///    their realized balance *increases* while their ownership of the pool *decreases*. To reflect this, we
-///    decrease their Shadow Balance, the Total Shadow Balance, their Real Balance, and the Total Real Balance.
 contract MixinStakingPoolRewards is
     IStakingEvents,
-    MixinDeploymentConstants,
     MixinConstants,
     MixinStorage,
     MixinScheduler,
     MixinStakingPoolRewardVault,
-    MixinStakingPool,
-    MixinTimeLockedStake,
     MixinStakeBalances
 {
 
     using LibSafeMath for uint256;
 
-    /// @dev Withdraws an amount in ETH of the reward for the pool operator.
-    /// @param poolId Unique id of pool.
-    /// @param amount The amount to withdraw.
-    function withdrawRewardForStakingPoolOperator(bytes32 poolId, uint256 amount)
-        external
-        onlyStakingPoolOperator(poolId)
-    {
-        _withdrawFromOperatorInStakingPoolRewardVault(poolId, amount);
-        poolById[poolId].operatorAddress.transfer(amount);
-    }
-
-    /// @dev Withdraws the total balance in ETH of the reward for the pool operator.
+    /// @dev Computes the reward balance in ETH of a specific member of a pool.
     /// @param poolId Unique id of pool.
-    /// @return The amount withdrawn.
-    function withdrawTotalRewardForStakingPoolOperator(bytes32 poolId)
-        external
-        onlyStakingPoolOperator(poolId)
-        returns (uint256)
+    /// @param member The member of the pool.
+    /// @return totalReward Balance in ETH.
+    function computeRewardBalanceOfDelegator(bytes32 poolId, address member)
+        public
+        view
+        returns (uint256 totalReward)
     {
-        uint256 amount = getBalanceOfOperatorInStakingPoolRewardVault(poolId);
-        _withdrawFromOperatorInStakingPoolRewardVault(poolId, amount);
-        poolById[poolId].operatorAddress.transfer(amount);
-
-        return amount;
+        // cache some values to reduce sloads
+        IStructs.DelayedBalance memory delegatedStake = delegatedStakeToPoolByOwner[member][poolId];
+        uint256 currentEpoch = getCurrentEpoch();
+
+        // value is always zero in these two scenarios:
+        //   1. The current epoch is zero: delegation begins at epoch 1
+        //   2. The owner's delegated is current as of this epoch: their rewards have been moved to the ETH vault.
+        if (currentEpoch == 0 || delegatedStake.lastStored == currentEpoch) return 0;
```

Comment:
These checks should probably be reversed, since we don't need `currentEpoch == 0` to be checked after epoch 0.

Reply:
Very good eye! I agree!